packages/edit-site/src/components/global-styles/font-library-modal/utils/index.js

```diff
@@ -135,39 +136,85 @@ export function getDisplaySrcFromFontFace( input, urlPrefix ) {
+export function makeFontFamilyFormData( fontFamily ) {
 	const formData = new FormData();
 	const { kebabCase } = unlock( componentsPrivateApis );
 
-	const newFontFamily = {
-		...fontFamily,
+	const { fontFace, category, ...familyWithValidParameters } = fontFamily;
+	const fontFamilySettings = {
+		...familyWithValidParameters,
 		slug: kebabCase( fontFamily.slug ),
 	};
 
-	if ( newFontFamily?.fontFace ) {
-		const newFontFaces = newFontFamily.fontFace.map(
-			( face, faceIndex ) => {
-				if ( face.file ) {
-					// Slugified file name because the it might contain spaces or characters treated differently on the server.
-					const fileId = `file-${ faceIndex }`;
-					// Add the files to the formData
-					formData.append( fileId, face.file, face.file.name );
-					// remove the file object from the face object the file is referenced by the uploadedFile key
-					const { file, ...faceWithoutFileProperty } = face;
-					const newFace = {
-						...faceWithoutFileProperty,
-						uploadedFile: fileId,
-					};
-					return newFace;
-				}
-				return face;
+	formData.append(
+		'font_family_settings',
+		JSON.stringify( fontFamilySettings )
+	);
+	return formData;
+}
+
+export function makeFontFacesFormData( font ) {
+	if ( font?.fontFace ) {
+		const fontFacesFormData = font.fontFace.map( ( face, faceIndex ) => {
+			const formData = new FormData();
+			if ( face.file ) {
+				// Slugified file name because the it might contain spaces or characters treated differently on the server.
+				const fileId = `file-${ faceIndex }`;
+				// Add the files to the formData
+				formData.append( fileId, face.file, face.file.name );
+				// remove the file object from the face object the file is referenced by the uploadedFile key
+				const { file, ...faceWithoutFileProperty } = face;
+				const fontFaceSettings = {
+					...faceWithoutFileProperty,
+					src: fileId,
+				};
+				formData.append(
+					'font_face_settings',
+					JSON.stringify( fontFaceSettings )
+				);
+			} else {
+				formData.append( 'font_face_settings', JSON.stringify( face ) );
 			}
-		);
-		newFontFamily.fontFace = newFontFaces;
+			return formData;
+		} );
+
+		return fontFacesFormData;
 	}
+}
 
-	formData.append( 'font_family_settings', JSON.stringify( newFontFamily ) );
-	return formData;
+export async function batchInstallFontFaces( fontFamilyId, fontFacesData ) {
+	const promises = fontFacesData.map( ( faceData ) =>
+		fetchInstallFontFace( fontFamilyId, faceData )
+	);
+	const responses = await Promise.allSettled( promises );
+
+	const results = {
+		errors: [],
+		successes: [],
+	};
+
+	responses.forEach( ( result, index ) => {
+		if ( result.status === 'fulfilled' ) {
+			const response = result.value;
+			if ( response.ok ) {
+				results.successes.push( response.data );
+			} else {
+				// Handle HTTP error statuses
+				results.errors.push( {
+					error: `HTTP error: ${ response.status }`,
+					data: fontFacesData[ index ],
```

Comment:
We want to know which font face has failed. Maybe this should be just font_face_settings, unless we want to do something with the file?

```suggestion
					data: fontFacesData[ index ].font_face_settings,
```